### DISCUSSION TITLE
Proposal for New Syntax and Features in GitHub Flavored Markdown

### DISCUSSION THREAD

**Discussion by AnthonyByansi** [community/community#57555] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Show & Tell

### Body

HEllo fellow committers, 

I would like to initiate a discussion regarding the integration of new syntax and features into **GitHub Flavored Markdown (GFM)**. I believe it is essential to consider expanding the capabilities of  **GFM.** 

As we strive to empower developers and foster collaboration, I believe it's crucial to explore ways to enhance the capabilities of GFM, making it the most appropriate Markdown language for our community

I Can't wait to share what I have and hope it will shape GFM in one way or the other.

> Let's collaborate and shape the future of  **GFM**  together.

**Comment by PeterWone** [comment]
I'd like to propose a change to the column alignment syntax for tables.

You've all seen this notation indicating a right-aligned column `|---:|`

I'd like to extend this to permit `[:0-9]` in the position currently occupied solely by the colon.

- All of these indicate right-alignment. 
- Digits indicate that when the column values are numbers they should be rounded and formatted to the specified number of decimal places.  `|-----2|` means 
       - 3.1 should be rendered as `3.10` 
       - 9.995 should be rendered as `10.00` 
       - 9.994 should be rendered as `9.99`

I considered the symbol $ indicating that when the column values are numbers they should be rounded to two decimal places and a currency symbol shown, however `$` will invite complaint from every nation that doesn't use dollars and sometimes people _don't_ want cents. It may be better just to support digits for number formatting, and statically specify the symbol in the heading.

Number formatting will stop rounding errors from making columns absurdly wide due to 15 decimal places (or whatever JS FP has).
